sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -97,6 +98,10 @@ def _format_url_section(template, **kwargs):
             formatted_components = template.split("/")
             components = [c for c in formatted_components if "{{{}}}".format(key.args[0]) not in c]
             template = "/".join(components)
+            if last_template == template:
```

Comment:
This feels complicated. Any way to make it simpler? Maybe something like:

```python
import collections

class Missing(dict):

    def __init__(self):
        self.touched = False

    def __missing__(self, key):
        self.touched = True
        return '\t'
    
def format_thing(template: str, **kwargs):
    unknowns = Missing()
    rendered = template.format_map(collections.ChainMap(kwargs, unknowns))
    if unknowns.touched:
        return rendered.replace('\t/', '')
    return rendered

print(format_thing('hello/world/{x}/{y}/{name}/{other}', name='foo'))
```

Reply:
@johanste 

When Python does the formatting for {aaa.bbb}, it looks for key "aaa" rather than "aaa.bbb".

If I run 

```py
import collections

class Missing(dict):

    def __init__(self):
        self.touched = False

    def __missing__(self, key):
        self.touched = True
        return '\t'
    
def format_thing(template: str, **kwargs):
    unknowns = Missing()
    rendered = template.format_map(collections.ChainMap(kwargs, unknowns))
    if unknowns.touched:
        return rendered.replace('\t/', '')
    return rendered

print(format_thing('hello/world/{x.x}/{y}/{name}/{other}', name='foo'))
```
It raised

AttributeError: 'str' object has no attribute 'x'.

it translates {x.x} into "\t.x".

Reply:
Taken offline.